rfc/composite_table_availability.pseudocode

```diff
@@ -0,0 +1,123 @@
+
+PhysicalTableSchema:
+	getFieldName(Column c)
+
+DataSource:
+	String getName()
+	Map<FieldName, Set<Segment>> raw segment data
+	Map<FieldName, List<Interval>> squashedFromSegments  // optional optimization
+
+	Set<FieldName> getFields() //union of all keys
+	List<Interval> getIntervals(FieldName fieldName)
+	Map<FieldName,List<Interval>> getIntervals(Set<FieldName>) // map slice from squashedMap
+	Set<SegmentId> getSegments(List<Interval>)  // Filter all segments by overlap on time
+
+DataSourceConstraints // Simple bean parameter object
+
+	getGrain
+	Set<FieldName> getMetrics
+	Set<Dimension> getDimensions
```

Comment:
For these, what level are we assuming the they are at? Logical level? Physical level? Are these already mapped down to their underlying Logical columns? Underlying _physical_ columns?